File: leetcode/217-ContainsDuplicate/217-ContainsDuplicate.java
// Last updated: 29/04/2026, 23:11:25
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int num: nums) {
            if(set.contains(num)) {
                return true;
            }
            set.add(num);
        }
        return false;
    }
}

File: leetcode/217-ContainsDuplicate/217-ContainsDuplicate.cpp
// Last updated: 29/04/2026, 23:11:32
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> duplicateChecker;

        for (int i = 0 ; i< nums.size(); i++) {
            if (duplicateChecker.find(nums[i]) != duplicateChecker.end()) {
                return true;
            }
            duplicateChecker.insert(nums[i]);
        }
        return false;
    }
};